### ISSUE TITLE
Option to have the server to disable movement/fall/(climb) bob

### ISSUE THREAD

**Issue by FatalError42O** [minetest/minetest#12416]
## Problem

It's impossible to have the player's camera line up with a object if needed for example, I'm working on a gun mod project, but if the player has movement bob it will make it impossible to aim while moving. Even with it off though, when a player "climbs" (gets really close to a block before jumping onto it's edge" it throws the camera further up, causing the gun to clip into the camera if aiming down sights! If it's not obvious, this is extremely frustrating and literally impossible to work around.

## Solutions

Some sort of API for disabling this is needed
maybe something like "player:set_view_bobbing_factor(amount)", "player:set_fall_bobing_factor(amount)"  (and for 'climb')
where 1 is of course what the player set it as etc, so it'd be a view bobbing factor-factor where it's something like:
bobbing*client bobbing factor*server bobbing factor

## Alternatives

There really aren't any alternative solutions I can think of.

## Additional context

I'd think that these API changes would be relatively easy to implement, if I knew C++ I would- but all I can do here is hope that this is fixed.

example of the issue bobbing can cause for mods:
(player while moving and aiming)
![image](https://user-images.githubusercontent.com/58855799/172504241-2cd334b8-aee8-4b5d-aeee-416aa14ea328.png)

**Comment by appgurueu** [comment]
Alternative: Tell your players to set "view bobbing factor" and "fall bobbing factor" to zero in the settings for best experience. Downside: No bobbing. Upsides: Accurate raycasts, visuals work. Thus related: https://github.com/minetest/minetest/issues/11227.

**Comment by FatalError42O** [comment]
> Alternative: Tell your players to set "view bobbing factor" and "fall bobbing factor" to zero in the settings for best experience. Downside: No bobbing. Upsides: Accurate raycasts, visuals work. Thus related: #11227.

I agree with the idea, but unless you plan on implementing it yourself i doubt this will happen in the near fututr

**Comment by FatalError42O** [comment]
Here's gameplay footage for why view bobbing sucks, and why we need an option to remove it.
https://youtu.be/V7ivox3edl0
